### PR TITLE
Add igxQuickFilterTemplate directive

### LINKED ISSUE #4299: igx-tree-grid filter design customization
## Is your feature request related to a problem? Please describe.
We have already using igtreegrid in angularjs application. now we are migrating to angular7 I need a normal filter design to integrate the grid. below attached screenshot shows sample design.
Also for some particular columns in the grid I have to use dropdown filter.

## Describe the solution you'd like
I need to customize igx-tree-grid filter design to our old style column filter design. also need dropdown filter for particular column

## Additional context
Sample design screenshot
teset

### PR DESCRIPTION
**Pull request by sstoyanovIG**
Closes #4299   

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [x] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by SAndreeva** [comment]
Retemplating works just fine but will not be applicable for Chrome until #4376 is fixed.

File: projects/igniteui-angular/src/lib/grids/column.component.ts [hunk 1]
```diff
@@ -632,6 +632,34 @@ export class IgxColumnComponent implements AfterContentInit {
     get bodyTemplate(): TemplateRef<any> {
         return this._bodyTemplate;
     }
+    /**
```
Comment (SAndreeva) [inline projects/igniteui-angular/src/lib/grids/column.component.ts]: The input is placed between the bodyTemplate getter and setter.

File: projects/igniteui-angular/src/lib/grids/column.component.ts [hunk 2]
```diff
@@ -718,6 +718,34 @@ export class IgxColumnComponent implements AfterContentInit {
...
         }
     }
+    /**
+     * Returns a reference to the `quickFilterTemplate`.
+     * ```typescript
+     * let quickFilterTemplate = this.column.quickFilterTemplate;
+     * ```
+     * @memberof IgxColumnComponent
+     */
+    @Input('quickFilterTemplate')
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grids/column.component.ts]: Could you please change the name to `filterCellTemplate`. It's a bit more meaningful for the users.

File: projects/igniteui-angular/src/lib/grids/grid.common.ts [hunk 3]
```diff
@@ -125,6 +125,13 @@ export class IgxColumnResizerDirective implements OnInit, OnDestroy {
     }
 }
 
+@Directive({
+    selector: '[igxQuickFilterTemplate]'
+})
+export class IgxQuickFilterTemplateDirective {
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grids/grid.common.ts]: Could you please change the name to `IgxFilterCellTemplateDirective`. It's a bit more meaningful for the users.

**Pull request merged**